jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/editor/CodeWhispererEditorListener.kt

```diff
@@ -18,6 +21,18 @@ class CodeWhispererEditorListener : EditorFactoryListener {
             object : DocumentListener {
                 override fun documentChanged(event: DocumentEvent) {
                     CodeWhispererInvocationStatus.getInstance().documentChanged()
+                    // When open a file for the first time, IDE will also emit DocumentEvent for loading with `isWholeTextReplaced = true`
+                    // Added this condition to filter out those events
+                    if (!event.isWholeTextReplaced) {
```

Comment:
question on this condition:
but what if user just replaced the whole text for example using their clipboard context? Do we still ignore them?

Reply:
If this case is property handled(i.e. do nothing) in `CodeWhispererCodeCoverageTracker` then we don't need this condition.

Reply:
> question on this condition: but what if user just replaced the whole text for example using their clipboard context? Do we still ignore them?

good point, let me experiment againt this scenario a bit

Reply:
> If this case is property handled(i.e. do nothing) in `CodeWhispererCodeCoverageTracker` then we don't need this condition.

sounds good, we can delegate the logic to codeCoverageTracker instead of doing these here

Reply:
updated